src/Control/Monad/Aff.purs

```diff
@@ -65,9 +65,12 @@ module Control.Monad.Aff
   cancelWith aff c = runFn3 _cancelWith nonCanceler aff c
 
   -- | Converts the asynchronous computation into a synchronous one. All values
-  -- | and errors are ignored.
-  launchAff :: forall e a. Aff e a -> Eff e Unit
-  launchAff = runAff (const (pure unit)) (const (pure unit))
+  -- | are ignored, and if the computation produces an error, it is thrown.
+  launchAff :: forall e a. Aff e a -> Eff (err :: EXCEPTION | e) Unit
```

Comment:
Or perhaps we could leave the definition like this, but just change the returned type, to, say `Eff (errUncatchable :: EXCEPTION | e) Unit`?